src/mapbox/mapbox.ts

```diff
@@ -852,7 +852,9 @@ export default class Mapbox {
         // Replace map.transform with ours during the callbacks
         map.transform = this._renderTransform;
         baseFire.call(map, event, properties);
+        const savedProjection = map.transform.projection;
         map.transform = tr;
+        map.transform.projection = savedProjection;
```

Comment:
Looks like the auto switching of projection happens in `map._render`:

https://github.com/mapbox/mapbox-gl-js/blob/9fff53107b78bdaae875693b7b34c7f9acb5a1ae/src/ui/map.js#L2950

Would it make more sense for us to match the two transforms in

https://github.com/visgl/react-map-gl/blob/bf5514908f653c60741136d02befefbadd0dff82/src/mapbox/mapbox.ts#L884

`tr.projection = this._map.transform.projection`?

Reply:
I tried that initially, and while it fixes the rendering issue, it introduces a bug where the map position jumps on zooming out past the Mercator/Globe threshold.

https://user-images.githubusercontent.com/14878684/165402326-332c9939-4c2f-4db1-9860-ff0aa809e0e9.mov

Reply:
I have seen the same issue in matching elevation between the two transforms. In addition to assigning

`tr.projection = this._renderTransform.projection`

You may also need to call `_calcMatrices()` on the transform.

Reply:
Thank you, I'll give that a try!